data-prepper-plugins/aws-lambda/src/main/java/org/opensearch/dataprepper/plugins/lambda/processor/LambdaProcessor.java

```diff
@@ -149,9 +136,25 @@ public Collection<Record<Event>> doExecute(Collection<Record<Event>> records) {
             return records;
         }
 
-        //lambda mutates event
+        // Initialize here to void multi-threading issues
+        // Note: By default, one instance of processor is created across threads.
+        BufferFactory bufferFactory = new InMemoryBufferFactory();
+        lambdaCommonHandler = new LambdaCommonHandler(LOG, lambdaAsyncClient, functionName, invocationType);
+        Buffer currentBufferPerBatch = lambdaCommonHandler.createBuffer(bufferFactory);
+        List futureList = new ArrayList<>();
+        totalFlushedEvents = 0;
+
+        // Setup request codec
+        JsonOutputCodecConfig jsonOutputCodecConfig = new JsonOutputCodecConfig();
```

Comment:
Can this config be moved to constructor?

Reply:
might need additional changes to tests, will work on it on a follow up PR.